src/main/java/com/github/fppt/jedismock/operations/hashes/HGetAll.java

```diff
@@ -23,22 +23,17 @@ protected Slice response() {
 
         Map<Slice, Slice> fieldAndValueMap = base().getFieldsAndValues(hash);
 
-        if(fieldAndValueMap == null) {
+        if (fieldAndValueMap == null) {
             fieldAndValueMap = new HashMap<>();
         }
-        int arraySize = fieldAndValueMap.size() * 2;
-        Slice [] fieldAndValueList = new Slice[arraySize];
 
-        int currentIndex = arraySize - 1;
-        for (Map.Entry<Slice, Slice> entry: fieldAndValueMap.entrySet()){
-            fieldAndValueList[currentIndex] = Response.bulkString(entry.getValue());
-            currentIndex--;
+        List<Slice> output = new ArrayList<>();
 
-            fieldAndValueList[currentIndex] = Response.bulkString(entry.getKey());
-            currentIndex--;
-        }
+        fieldAndValueMap.forEach((key, value) -> {
```

Comment:
Why adding them at zero position? Does order matter here? This will work as `O(n^2)` against the number of keys. Equivalent, but `O(n)` would be adding in a normal order and then reversing with `Collections.reverse`. However, I don't think that order matters

Reply:
Agree. I didn't want to change semantics so I saved the order.